lib/template/exec.go

```diff
@@ -506,6 +506,36 @@ func (s *state) walkRange(dot reflect.Value, r *parse.RangeNode) controlFlowSign
 			break
 		}
 		return controlFlowNone
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intVal := int(val.Int())
+		if intVal <= 0 {
+			break
+		}
```

Comment:
I would rather raise a runtime error in this case. WDYT?

Reply:
Sounds fine with me, but I think it might not be so user friendly if the value of `n` is a variable that the user does not know the value, like a count from DB
Perhaps `< 0` could be a runtime error but `= 0` could remain a simple break, WDYT?

Reply:
I think `range 0` should be permitted, but negative is nonsensical.

Reply:
Updated